### PR TITLE
Use docker cacert in SSLContext

### PR DESCRIPTION
**Pull request by csabapalfi**
You can manage the local cacert keystore with `keytool` on each host but when you deploy to production-like environments it's much simpler use a custom SSLContext which knows about the docker cacert.

### CODE REVIEW

File: docker/src/main/java/org/jclouds/docker/suppliers/SSLContextWithKeysSupplier.java [hunk 1]
```diff
@@ -35,36 +52,20 @@
...
    private final TrustManager[] trustManager;
    private final Supplier<Credentials> creds;
 
+
    @Inject
-   SSLContextWithKeysSupplier(@Provider Supplier<Credentials> creds, HttpUtils utils, TrustAllCerts trustAllCerts) {
-      this.trustManager = utils.trustAllCerts() ? new TrustManager[]{trustAllCerts} : null;
+   SSLContextWithKeysSupplier(@Provider Supplier<Credentials> creds, @Named(DockerApiMetadata.DOCKER_CA_CERT_PATH) String caCertPath) {
       this.creds = creds;
+      this.trustManager = !isNullOrEmpty(caCertPath) ? getTrustManagerWithCaCert(caCertPath) : null;
```
Comment (grkvlt) [inline docker/src/main/java/org/jclouds/docker/suppliers/SSLContextWithKeysSupplier.java]: Should we keep the original logic (to optionally use `TrustAllCerts`) if no CA certificate is supplied?

File: docker/src/main/java/org/jclouds/docker/suppliers/SSLContextWithKeysSupplier.java [hunk 2]
```diff
@@ -113,8 +128,8 @@ private static PrivateKey getKey(String privateKey) {
       }
    }
 
-   private static String loadFile(final String filePath) throws IOException {
-      return Files.toString(new File(filePath), Charsets.UTF_8);
+   private static String loadFile(final String filePath) throws IOException{
+         return Files.toString(new File(filePath), Charsets.UTF_8);
```
Comment (grkvlt) [inline docker/src/main/java/org/jclouds/docker/suppliers/SSLContextWithKeysSupplier.java]: Spacing is off here, I think

**Comment by csabapalfi** [comment]
Now part of #188

File: docker/src/main/java/org/jclouds/docker/suppliers/SSLContextWithKeysSupplier.java (hunk 2 above)
Comment (csabapalfi) [inline docker/src/main/java/org/jclouds/docker/suppliers/SSLContextWithKeysSupplier.java]: Fixing on `feature/docker-1.7`

File: docker/src/main/java/org/jclouds/docker/suppliers/SSLContextWithKeysSupplier.java (hunk 1 above)
Comment (csabapalfi) [inline docker/src/main/java/org/jclouds/docker/suppliers/SSLContextWithKeysSupplier.java]: Fixing on `feature/docker-1.7`

**Pull request closed without merging**